### PR TITLE
Feat/auth context

### PR DESCRIPTION
**Pull request by Viniciuslb067**
## Feature

---

**What I did:**

- remove react-query-auth
- implement own auth context

### CODE REVIEW

File: src/modules/incomes/hooks/useIncomes.tsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 
+import { getIncomes } from '../api';
+
 type UseIncomesOptions = {
   config?: QueryConfig<typeof getIncomes>;
 };
 
 export const useIncomes = ({ config }: UseIncomesOptions) => {
-  return useQuery('incomes', getIncomes, config);
+  return useQuery('incomes', getIncomes, config as any);
```
Comment (lucanevess87) [inline src/modules/incomes/hooks/useIncomes.tsx:12]: why config is typed with any?

File: src/modules/expenses/hooks/useExpenses.tsx [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
+import { getExpenses } from '../api';
+
 type UseExpensesOptions = {
   config?: QueryConfig<typeof getExpenses>;
 };
 
 export const useExpenses = ({ config }: UseExpensesOptions) => {
-  return useQuery('expenses', getExpenses, config);
+  return useQuery('expenses', getExpenses, config as any);
```
Comment (lucanevess87) [inline src/modules/expenses/hooks/useExpenses.tsx:12]: why config is typed any inside useQuery?

**Review by lucanevess87 — APPROVED** [review]

File: src/components/common/Link/Link.tsx [hunk 3]
```diff
@@ -9,7 +9,7 @@ type LinkProps =
 
 export const Link = ({ to, children, ...restProps }: LinkProps) => {
   return (
-    <ChakraLink as={RouterLink} to={to} {...restProps}>
+    <ChakraLink as={RouterLink} to={to} {...(restProps as any)}>
```
Comment (EdimarioJr) [inline src/components/common/Link/Link.tsx:12]: Why restProps as any in here?

File: src/lib/auth/context.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  useQuery,
+  useMutation,
+  UseMutateAsyncFunction,
+  QueryObserverResult,
+  RefetchOptions,
+} from 'react-query';
+
+import { queryClient } from '@/lib/react-query';
+
+export interface AuthProviderConfig<User = unknown, Error = unknown> {
```
Comment (EdimarioJr) [inline src/lib/auth/context.tsx:12]: The use of interfaces in here is to override the interfaces that are coming from the lib?

File: src/lib/auth/context.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    });
+
+    const logoutMutation = useMutation({
+      mutationFn: logoutFn,
+      onSuccess: () => {
+        queryClient.clear();
+      },
+    });
+
+    const value = useMemo(
```
Comment (EdimarioJr) [inline src/lib/auth/context.tsx:110]: Why the use of useMemo in here ?

File: src/lib/auth/context.tsx (hunk 4 above)
Comment (Viniciuslb067) [reply src/lib/auth/context.tsx:12]: i dont know what lib are you referring, but doesn't have this interface in others libs

File: src/lib/auth/context.tsx (hunk 5 above)
Comment (Viniciuslb067) [reply src/lib/auth/context.tsx:110]: the useMemo it's because it has some error or some update then will trigger some rerender and get the new data

**Pull request merged**